src/utils/index.ts

```diff
@@ -283,3 +285,28 @@ export function shallowEqual(objA: OAny, objB: OAny): boolean {
 
   return true
 }
+
+export function linkify(text: string): string {
+  const escaped = escapeHtml(text);
+  const linked = autoLink(escaped);
+  return linked;
+}
+
+const linkPattern = /(^|[\s\n]|<[A-Za-z]*\/?>)((?:(?:https?):\/\/|lichess\.org\/)[\-A-Z0-9+\u0026\u2019@#\/%?=()~_|!:,.;]*[\-A-Z0-9+\u0026@#\/%=~()_|])/gi;
+
+function linkReplace(_: string, before: string, url: string) {
+  const fullUrl = url.indexOf('http') === 0 ? url : 'https://' + url;
+  const minUrl = url.replace(/^(?:https:\/\/)?(.+)$/, '$1');
+  return before + '<a href="#" onClick="window.open(\'' + fullUrl + '\', \'_system\')">' + minUrl + "</a>";
+}
+
+const userPattern = /(^|[^\w@#/])@([\w-]{2,})/g
+
+function userLinkReplace(orig: string, prefix: String, user: string) {
+  if (user.length > 20) return orig;
+  return prefix + '<a href="#" onClick="window.open(\'' + globalConfig.apiEndPoint + '/@/' + user + '\', \'_system\')">@' + user + "</a>";
```

Comment:
User link should not open an external window, but user profile in the application.

Reply:
Here again, the user will be prompted as to whether they wish to open the link in a browser or the app. However, here it probably does make sense to just go directly to the user page in the app.